### DISCUSSION TITLE
Decomposition configuration files should support YAML, not only JSON

### DISCUSSION THREAD

**Discussion by morrone** [ovis-hpc/ldms#1656] [General]
We should be able to express decomposition configuration in YAML format, not just JSON.

JSON is not a good configuration file format, if for no other reason than it does not support comments. In a center with many different employees, comments are essential to help understanding why things are configured the way that they are.

**Comment by tom95858** [comment]
@morrone, @nick-enoent I think this is 'resolved'?

**Comment by morrone** [comment]
I don't think we can do decomp files in yaml yet, can we?

**Comment by tom95858** [comment]
Moved to v4.5.1

**Comment by emdonat** [comment]
Moved to a discussion to allow for planning of work and to scope need.

**Reply by morrone** [reply]
@emdonat can you please move it back to an issue? This is a task to be done. Not much discussion required. I understand dropping the v4.5.1 target, but this is not really a discussion topic.

**Reply by tom95858** [reply]
@morrone, no it is not a "task to be done." No alternative format has been proposed. This is a complaint like, "I don't like Mondays" Propose another day and the community can talk about it ... i.e. a discussion.

**Reply by morrone** [reply]
> @morrone, no it is not a "task to be done." No alternative format has been proposed. This is a complaint like, "I don't like Mondays" Propose another day and the community can talk about it ... i.e. a discussion.

The reason for the lack of proposed "alternative format" is because no changes are implied by this ticket.

YAML is a strict superset of JSON. We can do a direct one-to-one conversion of the JSON format to YAML. No changes needed. Hence, no changes proposed (at least in this ticket, format changes might be the subject of other tickets). No discussion required.

In fact, if done right, the existing json files should continue to work when the parser is YAML based. Because, again, YAML is a superset of JSON.
